x-pack/winlogbeat/module/sysmon/test/testdata/sysmon-11-filedelete.evtx.golden.json

```diff
@@ -70,5 +80,91 @@
       },
       "version": 5
     }
+  },
+  {
+    "@timestamp": "2020-05-12T06:48:27.084Z",
+    "event": {
+      "code": 23,
+      "kind": "event",
+      "module": "sysmon",
+      "provider": "Microsoft-Windows-Sysmon"
+    },
+    "fields": {
+      "event": {
+        "category": [
+          "file"
+        ],
+        "type": [
+          "deletion"
+        ]
+      }
+    },
+    "file": {
+      "directory": "C:\\Windows\\System32\\LogFiles\\Scm",
+      "name": "8b34f644-f627-47e7-98e0-957ba1c5eb6d",
+      "path": "C:\\Windows\\System32\\LogFiles\\Scm\\8b34f644-f627-47e7-98e0-957ba1c5eb6d"
+    },
+    "hash": {
+      "imphash": "00000000000000000000000000000000",
+      "md5": "5a9bddf83be530b481f0fd24db28a6ff"
+    },
+    "host": {
+      "name": "vagrant-2012-r2"
+    },
+    "log": {
+      "level": "information"
+    },
+    "process": {
+      "entity_id": "{42f11c3b-4664-5eba-91ae-000000000000}",
+      "executable": "C:\\Windows\\system32\\svchost.exe",
+      "hash": {
+        "imphash": "00000000000000000000000000000000",
```

Comment:
Maybe we should drop these? @andrewstucki, does this value have any meaning?

Reply:
So that looks odd to me--my guess is that sysmon is returning all zeroes in the case of not being able to grab the import symbols of the binary--i'd have to read the docs around it a bit more. That said, for sure that's not a valid value. Also--`imphash` itself is a `pe` specific thing and shouldn't be treated like other general purposes hashes here, we actually carved out a field for it in `pe` in ECS--so this should get:

1. Dropped if it's all zeroes
2. Potentially dropped from the `related.hash` field (clarifying with @webmat that that's meant for general purpose hashes, correct?)
3. Moved from `process.hash.imphash` to `process.pe.imphash`

Reply:
btw, realized that imphash stuff is actually still not fully released (it's in unreleased ECS 1.6). It's not going away any time soon since the endpoint is using this field as well, but just making sure we're cool shipping this field a bit early?

https://www.elastic.co/guide/en/ecs/master/ecs-pe.html

Reply:
Semantics of `related.hash` are very broad. Any hash should be added there, even if it's a very specific thing like imphash 👍

Reply:
then, keeping it in `related.hash` and moving it from `process.hash.imphash` to `process.pe.imphash` sounds good, as far as it is good to ship `process.pe.imphash` now?

Reply:
Forward compatibility on upcoming fields like `process.pe.imphash` is a product decision.

Although from the schema perspective, I've seen zero controversy about this new field. So I can say the risks of it changing until the release are very slim. So this has my 👍

Reply:
I think so.

For testing this I'd try seeing if you can ensure that bad `imphash` isn't in the output--and then try an generate another event that actually captures the proper imphash.

I poked around to find references to sysmon returning all zero hashes and [this is the only thing I could find](https://github.com/Neo23x0/sigma/blob/master/rules/windows/sysmon/sysmon_ads_executable.yml#L22) and it seems to indicate an "empty imphash value".

What's interesting is that I just grabbed the hash for `svchost.exe` from the Windows 2019 VM and it should be `247b9220e5d9b720a82b2c8b5069ad69`--so I'm wondering if something is just misconfigured in sysmon? Either way we should be stripping out the 0000s--so a test case for inclusion of valid values and exclusion of the all zeros might be nice.